src/main/java/frc/robot/subsystems/ColorPanelSubsystem.java

```diff
@@ -0,0 +1,148 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorMatch;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import frc.robot.posev2.WheelColorValue;
+import edu.wpi.first.wpilibj.DriverStation;
+
+public class ColorPanelSubsystem extends BaseSubsystem{
+    private final I2C.Port i2cPort = I2C.Port.kOnboard;
+    private final ColorSensorV3  colorSensor = new ColorSensorV3(i2cPort);
+    private final ColorMatch  colorMatcher = new ColorMatch();
+    private final Color kBlueTarget = ColorMatch.makeColor(0.143, 0.427, 0.429);
+    private final Color kGreenTarget = ColorMatch.makeColor(0.197, 0.561, 0.240);
+    private final Color kRedTarget = ColorMatch.makeColor(0.561, 0.232, 0.114);
+    private final Color kYellowTarget = ColorMatch.makeColor(0.361, 0.524, 0.113);
+
+    @Override
+    public void initialize() {
+         colorMatcher.addColorMatch(kBlueTarget);
+         colorMatcher.addColorMatch(kGreenTarget);
+         colorMatcher.addColorMatch(kRedTarget);
+         colorMatcher.addColorMatch(kYellowTarget);  
+    }
+    public void detectColor(){
```

Comment:
this method never provides a way to get the data. It doesnt update state, and it doesnt return a value.
Also, detectColor could be named better-- is it geting the color the robot sees, or the field sees? or what the field wants?